sdk/test-utils/recorder/src/createRecorder.ts

```diff
@@ -38,6 +39,7 @@ export class NockRecorder extends BaseRecorder {
       this.relativeTestRecordingFilePath,
       testFilePath
     ).testInfo;
+    mockMsalAuth(nock);
```

Comment:
Please add comments. These is the kind of changes that make the recorder far less generic. It may be hard to clean this up if eventually we don't need MSAL.

Also, I don't have it clear if this is necessary every time, for every recording.

The fact that this is to solve 4 failing tests that would be fixed with re-recording, makes me unsure about this change.

On the other hand, I like the idea of fixing this once and for all.

I think it's better to keep this, but please comment this so that, imagine, some years later somebody can understand more clearly whether removing this or changing this may have what impact.

Reply:
> Please add comments. These is the kind of changes that make the recorder far less generic. It may be hard to clean this up if eventually we don't need MSAL.

Will add comments, it won't break anything though even if we switch from msal.

> Also, I don't have it clear if this is necessary every time, for every recording.

Needed for all the AAD tests.

> The fact that this is to solve 4 failing tests that would be fixed with re-recording, makes me unsure about this change.

No, re-recording won't fix. This is for every test that is using AAD. Remember the difference in request body that I pointed at 
https://github.com/Azure/azure-sdk-for-js/pull/14993#issuecomment-825334524
![image](https://user-images.githubusercontent.com/10452642/117722350-ccd9c780-b195-11eb-836e-ce6fd26b72d1.png)

Reply:
Alright, documentation will be key to make this easier for us over time then!